### PR TITLE
(issue-#24) removing defer

### PR DESCRIPTION
**Pull request by khushbuparakh**
Signed-off-by: khushbuparakh <khushbuparakh@hotmail.com>
I started from ReadAt, WriteAt functions.
So func (s *Server) had a defer code. 
I modified it to return err
Then used at the very low level I/O code and 
follow the path to know IO path in controller side, and,
similarly Handle() in rpc/server.go for replica side IO handling.

### CODE REVIEW

File: backend/remote/remote.go [hunk 1]
```diff
@@ -86,12 +86,12 @@ func (r *Remote) doAction(action string, obj interface{}) error {
...
 		return err
 	}
-	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
+		resp.Body.Close()
 		return fmt.Errorf("Bad status: %d %s", resp.StatusCode, resp.Status)
 	}
-
+    resp.Body.Close()
```
Comment (utkarshmani1997) [inline backend/remote/remote.go]: @khushbuparakh  can you fix the indentation ?

File: controller/control.go [hunk 2]
```diff
@@ -208,15 +208,16 @@ func (c *Controller) SetReplicaMode(address string, mode types.Mode) error {
 	switch mode {
 	case types.ERR:
 		c.Lock()
-		defer c.Unlock()
+		c.setReplicaModeNoLock(address, mode)
+		c.Unlock()
 	case types.RW:
 		c.RLock()
-		defer c.RUnlock()
+		c.setReplicaModeNoLock(address, mode)
```
Comment (utkarshmani1997) [inline controller/control.go:215]: why are you adding c.setReplicaModeNoLock(address, mode) at these places. It's not part of defer function.

File: controller/control.go [hunk 3]
```diff
@@ -304,7 +306,6 @@ func (c *Controller) Start(addresses ...string) error {
 			return err
 		}
 		// We will validate this later
-		c.setReplicaModeNoLock(address, types.RW)
 	}
```
Comment (utkarshmani1997) [inline controller/control.go:309]: Why this code has been removed ?

**Review by utkarshmani1997 — COMMENTED** [review]
Thanks for the PR. changes are looking good but not sure of couple of things so i have added few review comments.

File: controller/control.go (hunk 2 above)
Comment (khushbuparakh) [reply controller/control.go:215]: @utkarshmani1997 It's, in the end, I just move it to defer unlock statement. If you check the line 219 you will understand why I added it here

Comment (utkarshmani1997) [reply controller/control.go:215]: ok, sounds good.

**Comment by khushbuparakh** [comment]
Hi @utkarshmani1997 can you re review. I will also make another PR in Jiva repo

**Review by utkarshmani1997 — APPROVED** [review]
Thanks @khushbuparakh for this PR. Changes look good to me. @payes kindly review this PR and merge.

**Pull request closed without merging**